### ISSUE TITLE
Ability to add custom icons

### ISSUE THREAD

**Issue by fullcavediver** [helloworld1/FreeOTPPlus#130]
Please add the ability to add custom icons or add the following services to the library:
- netcloud
- gmx (german mail service provider gmx.net)
- wiki.js
Thank you!

**Comment by carlphilipp** [comment]
It is something that has been bothering me a little bit too. If the maintainers are ok, I can fix that. Here is what I'm thinking about. Current version:
![image](https://user-images.githubusercontent.com/818104/117742026-6a9aba00-b1c9-11eb-8489-1ca0347c1d25.png)
What I can do:
![image](https://user-images.githubusercontent.com/818104/117742101-95850e00-b1c9-11eb-95cd-7ebcf9a2b7b0.png)

Would that be ok?

**Comment by helloworld1** [comment]
The gray theme was added to improve the contrast of some dark icons. If you make the background too dark, some darker icons may not be visible. The correct fix would be to make background adaptable to the icon.
https://github.com/helloworld1/FreeOTPPlus/issues/63

**Comment by fullcavediver** [comment]
> It is something that has been bothering me a little bit too. If the maintainers are ok, I can fix that. Here is what I'm thinking about. Current version:
> ![image](https://user-images.githubusercontent.com/818104/117742026-6a9aba00-b1c9-11eb-8489-1ca0347c1d25.png)
> What I can do:
> ![image](https://user-images.githubusercontent.com/818104/117742101-95850e00-b1c9-11eb-95cd-7ebcf9a2b7b0.png)
> 
> Would that be ok?

Hi Carl-Philipp, yes, your proposal is perfect! That's exactly how I imagined it to be.

**Comment by fullcavediver** [comment]
I have another suggestion - themes. I know it's a lot more work, but design is the second most important attribute besides functions. You can build in 3 themes - light, dark and custom with the options: text color, background color, primary color,... simply the most important colors to choose from

**Comment by carlphilipp** [comment]
@helloworld1 I'm not a UI person, but it sounds like what you are describing would be with having different background depending on the actual icon. And I'm not sure that would be very attractive visually. My proposition was more around having the same background but a dark version of the icon if needed. That's how I have seen it done in other apps

**Comment by fullcavediver** [comment]
I also think that the undepending black background for all icons is much nicer than an adaptive one. Everyone always has the possibility to choice an optimal icon

**Comment by fullcavediver** [comment]
[https://play-lh.googleusercontent.com/nkXDb1QsWgN5visUFYmYbYVTwaLDHcVsp0rbF8GnX803H4LRQNK5_SiA9Mh4aZTuDJiY=w1920-h937-rw](url)
Do you think this design is elegant?

**Comment by helloworld1** [comment]
Yes it does look great if all icons are not using black elements. Unfortunately some icons are designed only for light theme. Maybe another option is fixing all those icons to use lighter gray so it's compatible with both light and dark theme.

**Comment by mrfuggy** [comment]
Also, three dot button (edit/delete menu) is difficult to see in standard and dark modes. Three dots are displayed in black on dark gray. The button is hard to find.

**Comment by praecipitator** [comment]
Goddammit, this is over 2 years old now? I guess this won't be fixed. Too bad the dark theme can't be turned off anymore if the device itself is set to dark mode...
